docs/web-component.md

```diff
@@ -0,0 +1,888 @@
+    template.innerHTML = `<section><p>Hello World!</p></section>`;
+
+    const templateBtn = document.createElement('template');
+    templateBtn.innerHTML = '<button>Click me!</button>';
+
+    this._shadowRoot = this.attachShadow({
+      mode: 'open',
+      delegatesFocus: false
+    });
+    this._shadowRoot.appendChild(template.content.cloneNode(true));
+    this._shadowRoot.appendChild(templateBtn.content.cloneNode(true));
+
+    this.$paragraph = this._shadowRoot.querySelector('p');
+    this.$button = this._shadowRoot.querySelector('button');
+    this.$button.addEventListener('click', () => {
+      if (this.LuigiClient) {
+        this.LuigiClient.uxManager().showAlert({
+          text: 'Hello from uxManager in Web Component',
+          type: 'info'
+        });
+      }
+    });
+  }
+
+  set context(ctx) {
+    this.$paragraph.innerHTML = ctx.title;
+  }
+}
+```
+
+As you could notice from previous example, you can use a LuigiClient instance inside your  web component class. 
+It is really important to notice, that this LuigiClient instance is different from the one than you can find in [client library](https://docs.luigi-project.io/docs/luigi-client-setup).
+
+Normal micro frontends are embedded inside iFrame: Luigi offers a library to allow the frontend to communicate with Luigi Core.
+In Web Component the situation is quite different: they are not encapsulated into an iframe, they are just loaded inside a shadow element; 
+when Luigi Core loads a Web Component, it injects a LuigiClient instance.
+
+### Luigi Client for web component
+
+In this javascript object, you can basically find two elements:
+- this.LuigiClient.uxManager(): you can use all methods described in [Luigi Core UX](https://docs.luigi-project.io/docs/luigi-core-api?section=ux)
+- this.LuigiClient.linkManager(): you can use all methods described in [Luigi Navigation](https://docs.luigi-project.io/docs/luigi-core-api?section=luiginavigation)
+
+For example, if you want to open an alert notificatoin, you can easily call:
+
+
+Here below, a very easy Hello World web component example:
+```javascript
+export default class extends HTMLElement {
+  constructor() {
+    ........
+    this.LuigiClient.uxManager().showAlert({
+      text: 'Hello from uxManager in Web Component',
+      type: 'info'
+    });
+    ........ 
+  }
+}
+```
+
+If you want open a drawer:
+```javascript
+export default class extends HTMLElement {
+  constructor() {
+    ........
+     this.LuigiClient.linkManager().openAsDrawer('Your Drawer Url', {header:true, backdrop:true, size:'s'});
+    ........ 
+  }
+}
+```
+
+
+## Tip: how to inject HTML Template code in web component 
+
+Sometime your Web Component has some HTML template that you would like to use, instead of creating dom elements one by one.
+We suggest putting your html template inside a variable at the beginning of the js file and append to the web component root in the constructor, here an example:
+
```

Comment:
```suggestion
Sometimes your Web Component has some HTML template that you would like to use, instead of creating DOM elements one by one.
We suggest putting your HTML template inside a variable at the beginning of the js file and append to the web component root in the constructor. An example is given below:
```